solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/inbound/JCSMPInboundChannelAdapter.java

```diff
@@ -34,352 +31,361 @@
+                                      @Nullable SolaceMeterAccessor solaceMeterAccessor) {
+        this.consumerDestination = consumerDestination;
+        this.jcsmpSession = jcsmpSession;
+        this.consumerProperties = consumerProperties;
+        this.endpointProperties = endpointProperties;
+        this.solaceMeterAccessor = solaceMeterAccessor;
+        this.flowReceivers = new ArrayList<>(consumerProperties.getConcurrency());
+        this.consumerStopFlags = new HashSet<>(consumerProperties.getConcurrency());
+    }
+
+    @Override
+    protected void doStart() {
+        final String endpointName = consumerDestination.getName();
+        logger.info(String.format("Creating %s consumer flows for %s %s <inbound adapter %s>",
+                consumerProperties.getExtension().getEndpointType(),
+                consumerProperties.getConcurrency(), endpointName, id));
+
+        if (isRunning()) {
+            logger.warn(String.format("Nothing to do. Inbound message channel adapter %s is already running", id));
+            return;
+        }
+
+        if (consumerProperties.getConcurrency() < 1) {
+            String msg = String.format("Concurrency must be greater than 0, was %s <inbound adapter %s>",
+                    consumerProperties.getConcurrency(), id);
+            logger.warn(msg);
+            throw new MessagingException(msg);
+        }
+
+        if (jcsmpSession instanceof JCSMPBasicSession jcsmpBasicSession
+                && !jcsmpBasicSession.isRequiredSettlementCapable(
+                Set.of(Outcome.ACCEPTED, Outcome.FAILED, Outcome.REJECTED))) {
+            String msg = String.format("The Solace PubSub+ Broker doesn't support message NACK capability, <inbound adapter %s>", id);
+            throw new MessagingException(msg);
+        }
+
+        if (executorService != null && !executorService.isTerminated()) {
+            logger.warn(String.format("Unexpectedly found running executor service while starting inbound adapter %s, " +
+                    "closing it...", id));
+            stopAllConsumers();
+        }
+
+        EndpointType endpointType = consumerProperties.getExtension().getEndpointType();
+        EndpointProvider<?> endpointProvider = EndpointProvider.from(endpointType);
+
+        if (endpointProvider == null) {
+            String msg = String.format("Consumer not supported for destination type %s <inbound adapter %s>",
+                    consumerProperties.getExtension().getEndpointType(), id);
+            logger.warn(msg);
+            throw new MessagingException(msg);
+        }
+
+
+        Endpoint endpoint = null;
+        try {
+            endpoint = (EndpointType.TOPIC_ENDPOINT.equals(endpointType) && consumerDestination.isTemporary()) ?
+                    endpointProvider.createTemporaryEndpoint(endpointName, jcsmpSession)
+                    : endpointProvider.createInstance(endpointName);
```

Comment:
For future us, can you add quick a comment to the code here about why you're doing this?

I assume, this is this done this way because `createTemporaryEndpoint(String,JCSMPSession)` for `EndpointType.QUEUE` adds a generated prefix in front of the name? But the `endpointName` already contains the full queue name (including the generated prefix). So running `endpointProvider.createTemporaryEndpoint()` again would have added the generated prefix twice? Is my understanding of this correct?

Reply:
Yes, exactly